.github/workflows/build-and-run-tests-posix.yml

```diff
@@ -0,0 +1,65 @@
+name: Build and run tests on Linux and MacOS platforms
+
+on:
+  workflow_call:
+    inputs:
+      platform:
+        required: true
+        type: string
+
+jobs:
+  build-and-run-tests-posix:
+    runs-on: ${{ inputs.platform }}
+    steps:
+    - run: echo "Job triggered by ${{ github.event_name }} event, running on ${{ runner.os }}."
+    - run: echo "Branch ${{ github.ref }} on repository ${{ github.repository }}"
+
+    - name: Check out repository code
+      uses: actions/checkout@v3
+
+    - name: Install Ninja
+      run: |
+        if [[ "${{ inputs.platform }}" == ubuntu-latest ]]; then
+          sudo apt-get install ninja-build
+        elif [[ "${{ inputs.platform }}" == macos-latest ]]; then
+          brew install ninja
+        else
+          echo Do not know how to install Ninja on platform ${{ inputs.platform }}
+          false
+        fi
+      shell: bash
+
+    - name: Retrieve Meson from cache
+      id: cache-meson
+      uses: actions/cache@v3
+      with:
+        path: meson-0.63.0
+        key: meson-0.63.0
+
+    - name: Download Meson
+      if: steps.cache-meson.outputs.cache-hit != 'true'
+      run: wget -q https://github.com/mesonbuild/meson/releases/download/0.63.0/meson-0.63.0.tar.gz &&
```

Comment:
For me, this was the least complicated way to make it work. Can we be sure that robinraju/release-downloader is more stable than the URL?

I have no strong opinion on this, but for now I am lazy and won't change it. Let me know if you think this is important.

Reply:
I think it is not important (for now), but the main power of github actions are the shared actions from all over; and it is always good to look if there are solutions that exist (and are maintained) before one hand crafts a single purpose solution.

These are actions I use:
* actions/checkout
* actions/upload-artifact
* actions/download-artifact
* actions/github-script
* EndBug/add-and-commit
* EndBug/latest-tag
* Bhacaz/checkout-files
* softprops/action-gh-release
* robinraju/release-downloader
* octokit/request-action
* mukunku/tag-exists-action
* JamesIves/github-pages-deploy-action

I tried to implement some for gitLAB for Tim, and that is a nightmare. Searching the actions 'shop' and reusing stuff from there is usually helpful, because more corner cases are covered. And if in doubt you can
1. look into the code
2. prescribe an exact version to use